### PR TITLE
Add `ingest` information to the cluster info endpoint

### LINKED ISSUE #95392: Extract ingest stats from nodes stats into separate API
### Description

From the nodes stats API, we should extract the `ingest` stats into their own endpoint. This endpoint can then be non-node-centric, and present a global view of the pipelines, processors, and the metrics collected around ingest in general.

### PR DESCRIPTION
**Pull request by HiDAl**
Add a new target (`ingest`) to the `/_info` API. It consolidates all the `ingest` information from the cluster nodes and returns a summary at the cluster level (compared with `_nodes/stats/ingest` it lacks the `<node>` dimension)

```json
GET _info/ingest
{
  "cluster_name": "runTask",
  "ingest": {
    "total": {
      "count": 77,
      "time_in_millis": 36,
      "current": 0,
      "failed": 0
    },
    "pipelines": {
      "test": {
        "count": 77,
        "time_in_millis": 4,
        "current": 0,
        "failed": 0,
        "processors": [
          {
            "set": {
              "type": "set",
              "stats": {
                "count": 12,
                "time_in_millis": 0,
                "current": 0,
                "failed": 0
              }
            }
          },
          {
            "set": {
              "type": "set",
              "stats": {
                "count": 12,
                "time_in_millis": 0,
                "current": 0,
                "failed": 0
              }
            }
          },
          {
            "remove": {
              "type": "remove",
              "stats": {
                "count": 12,
                "time_in_millis": 0,
                "current": 0,
                "failed": 0
              }
            }
          }
        ]
      },
      ...
   }
  }
}

GET _info/_all
{
  "cluster_name": "runTask",
  "http": {
    "current_open": 1,
    "total_opened": 112,
    "clients": [
      {
        "id": 873375542,
        "local_address": "127.0.0.1:9200",
        "remote_address": "127.0.0.1:59560",
        "last_uri": "/_info/ingest",
        "opened_time_millis": 1684937431630,
        "closed_time_millis": 1684937431630,
        "last_request_time_millis": 1684937431630,
        "request_count": 1,
        "request_size_bytes": 0
      },
      {
        "id": 1090681032,
        "local_address": "127.0.0.1:9200",
        "remote_address": "127.0.0.1:59575",
        "last_uri": "/",
        "opened_time_millis": 1684937474976,
        "closed_time_millis": 1684937474976,
        "last_request_time_millis": 1684937474976,
        "request_count": 1,
        "request_size_bytes": 0
      },
      {
        "id": 2018590882,
        "local_address": "127.0.0.1:9200",
        "remote_address": "127.0.0.1:59576",
        "last_uri": "/_info/_all",
        "opened_time_millis": 1684937475181,
        "last_request_time_millis": 1684937475181,
        "request_count": 1,
        "request_size_bytes": 0
      },
      ...
    ]
  },
  "ingest": {
    "total": {
      "count": 77,
      "time_in_millis": 36,
      "current": 0,
      "failed": 0
    },
    "pipelines": {
      "test": {
        "count": 77,
        "time_in_millis": 4,
        "current": 0,
        "failed": 0,
        "processors": [
          {
            "set": {
              "type": "set",
              "stats": {
                "count": 12,
                "time_in_millis": 0,
                "current": 0,
                "failed": 0
              }
            }
          },
          {
            "set": {
              "type": "set",
              "stats": {
                "count": 12,
                "time_in_millis": 0,
                "current": 0,
                "failed": 0
              }
            }
          },
          {
            "remove": {
              "type": "remove",
              "stats": {
                "count": 12,
                "time_in_millis": 0,
                "current": 0,
                "failed": 0
              }
            }
          }
        ]
      },
      ...
    }
  }
}
```

closes #95392

### CODE REVIEW

File: docs/reference/cluster/cluster-info.asciidoc [hunk 1]
```diff
@@ -126,17 +131,130 @@ Cumulative size in bytes of all requests from this client.
...
+`total`::
+(object)
+Contains information about ingest operations for the cluster.
++
+.Properties of `total`
+[%collapsible%open]
+=======
+`count`::
+(integer)
+Total number of documents ingested during the lifetime of the cluster.
```
Comment (masseyke) [inline docs/reference/cluster/cluster-info.asciidoc:151]: Is it the lifetime of the cluster? Or the lifetimes of all the nodes that are currently in the cluster (i.e. a node loses these counts on restart right?)?

Comment (HiDAl) [reply docs/reference/cluster/cluster-info.asciidoc:151]: Right... Will see how to reword this... Anyway, this is going to be a really weird number, because it won't be all the time increasing with nodes restarting... 
@dakrone This is an important point to consider for these statistics.

File: docs/reference/cluster/cluster-info.asciidoc [hunk 2]
```diff
@@ -126,17 +131,130 @@ Cumulative size in bytes of all requests from this client.
...
-[source,console]
-----
 # returns the http info of the cluster
 GET /_info/http
+
+# returns the http info of the cluster
+GET /_info/ingest
+
+# returns the http and ingest info of the cluster
+GET /_info/ingest,ingest
```
Comment (masseyke) [inline docs/reference/cluster/cluster-info.asciidoc:259]: I think you meant `GET /_info/http,ingest`

**Review by masseyke — COMMENTED** [review]
This looks pretty good to me. Tagging @joegallo in case he's currently working on something that would impact this.

File: docs/reference/cluster/cluster-info.asciidoc (hunk 2 above)
Comment (HiDAl) [reply docs/reference/cluster/cluster-info.asciidoc]: thanks!

File: modules/ingest-common/src/yamlRestTest/resources/rest-api-spec/test/ingest/15_info_ingest.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      cluster.info:
+        target: [ ingest ]
+
+  - is_true: cluster_name
+
+  # Summary ingest section
+  - is_true: ingest.total
+  - gte: { ingest.total.count: 2 }
+  - gte: { ingest.total.time_in_millis: 0 }
+  # next 2 conditions _should_ be 0, but because these yaml tests are sharing the same test cluster, other tests could
```
Comment (joegallo) [inline modules/ingest-common/src/yamlRestTest/resources/rest-api-spec/test/ingest/15_info_ingest.yml:57]: Minor nit, but the issue in this comment applies to all these `gte`s, not just the two below. Maybe reword and relocate?

**Comment by joegallo** [comment]
Not a review, just a comment, but this new capability puts a song in my heart. 😄

**Comment by HiDAl** [comment]
@elasticsearchmachine run elasticsearch-ci-docs

**Comment by HiDAl** [comment]
@elasticsearchmachine run elasticsearch-ci/docs

**Review by masseyke — APPROVED** [review]

File: docs/reference/cluster/cluster-info.asciidoc (hunk 1 above)
Comment (joegallo) [reply docs/reference/cluster/cluster-info.asciidoc]: Indeed, this is tricky. Each node tracks the information it has seen since it started. This cluster-wide info view is the sum of those since-the-node-started statistics for all the nodes.

It's straightforward enough to talk about on this PR, but I'm not sure what I think would be appropriate wording for documentation. 😬
